query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -111,6 +112,9 @@ public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheN
    }
 
    private void addCacheDependencyIfNeeded(String cacheStarting, EmbeddedCacheManager cacheManager, Properties properties) {
+      if (!ClusterRegistryImpl.GLOBAL_REGISTRY_CACHE_NAME.equals(cacheStarting)) {
+         cacheManager.addCacheDependency(cacheStarting, ClusterRegistryImpl.GLOBAL_REGISTRY_CACHE_NAME);
+      }
```

Comment:
I know this is not fixing the issue. The real problem is QueryFacadeImpl starts to create a query without going through the normal route of SearchManager.getQuery(..) so that creates plenty of chances the target class was not yet enabled via QueryInterceptor.enableClasses() in the first place. ClusterRegistry consistency might be a second cause. Please ignore this PR until I sort it out. (I've just created it to piggyback on the CI). Sorry for the noise.